src/zone.js

```diff
@@ -593,6 +594,10 @@ class Zone extends common.ServiceObject {
       },
       config
     );
+    if (body.sourceInstanceTemplate) {
```

Comment:
Do you like the idea of mapping this to `config.template`? The `createVM()` method is trying to be very simple with the terminology, since the Compute API is so complex. I don't think there's another "template" concept the API would expect in this context, so the short hand should be easier to remember and intuitive from this layer.

Reply:
I'm quite open for other suggestions, The reason I've chosen this way of going forward is since this is how I've seen it in other parts of this repository, and for the sake of keeping consistency, feels quite natural on my eyes to go this way.
Also I would expect something like this to pass the template id, but once again feel free to change it to something that you feel would be a better approach.

Reply:
Just to make sure I was being clear, I do think we should support the user passing a template reference for the request, I just thought `template` would be an easier name on the user to remember/type than `sourceInstanceTemplate`. As an example of similar shortcuts, `config.http` expands to `{tags: {items: ['http-server']}}`. Do you still think the full name the API recognizes is better? (I'm not convinced either way, just honestly interested)

Reply:
Oh, I didn't understand it properly.
In this case, yes. I do agree with you. I'll push the changes with what you have mentioned :)